pkg/collector/corechecks/cluster/orchestrator/collectors/inventory.go

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver && orchestrator
+// +build kubeapiserver,orchestrator
+
+package collectors
+
+import "fmt"
+
+// CollectorInventory is used to store and retrieve available collectors.
+type CollectorInventory struct {
+	collectors []Collector
+}
+
+// NewCollectorInventory returns a new inventory containing all known
+// collectors.
+func NewCollectorInventory() *CollectorInventory {
+	return &CollectorInventory{
+		collectors: []Collector{
+			newK8sClusterCollector(),
```

Comment:
I did this because:
- the inventory is not a singleton
- we get either multiple collectors by name or the list of all stable collectors only once when the bundle is initialized (i.e. during the orchestrator check configure step).

So not maintaining a map of collector names or other collector properties would still be fine because it's not a hot path.

Reply:
yeah it is not a hot path so performance wouldn't be a reason to refactor indeed 👍